packages/editor/src/components/browser/TypeBrowser.tsx

```diff
@@ -13,40 +14,99 @@ export const useTypeBrowser = (onDoubleClick: () => void): UseBrowserImplReturnV
     id: TYPE_BROWSER_ID,
     name: 'Type',
     content: <TypeBrowser value={value} onChange={setValue} onDoubleClick={onDoubleClick} />,
-    accept: () => value
+    accept: () => value,
+    icon: IvyIcons.DataClass
   };
 };
 
+type TypeKind = 'DATACLASS' | 'NORMALTYPE' | 'OWNTYPE';
+
 const TypeBrowser = (props: { value: string; onChange: (value: string) => void; onDoubleClick: () => void }) => {
+  type TypeBrowserObject = JavaType & {
+    kind: TypeKind;
+  };
+
   const { context } = useEditorContext();
 
+  const [allSearchActive, setAllSearchActive] = useState<boolean>(false);

```

Comment:
<boolean> is not needed if you set the default value to false -> `useState(false)`